File: backend/src/main/java/com/finwise/service/GoalService.java
package com.finwise.service;

import com.finwise.dto.goal.GoalRequest;
import com.finwise.dto.goal.GoalResponse;
import java.math.BigDecimal;
import java.util.List;

public interface GoalService {
    List<GoalResponse> getAllByUser(String userId);
    GoalResponse getById(String id);
    GoalResponse create(String userId, GoalRequest request);
    GoalResponse update(String id, GoalRequest request);
    void delete(String id);
    GoalResponse addContribution(String id, BigDecimal amount);
}

File: frontend/src/services/goalService.ts
import api from './api';

export const goalService = {
  getAll: () => api.get('/goals'),
  getById: (id: string) => api.get(`/goals/${id}`),
  create: (data: Record<string, unknown>) => api.post('/goals', data),
  update: (id: string, data: Record<string, unknown>) => api.put(`/goals/${id}`, data),
  delete: (id: string) => api.delete(`/goals/${id}`),
  addContribution: (id: string, amount: number) => api.post(`/goals/${id}/contribute`, { amount }),
};
